### PR TITLE
[AIDAPP-90] Review Open Service Requests query logic for dashboard

### PR DESCRIPTION
**Pull request by ankit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-90

### Technical Description

> Describe your changes in detail. This should include technical/architectural decisions and reasoning, if applicable.
> Added more status (except closed) for open service requests count in widget. Fix height issue in Line chart, Donut chart and they have now equal height.
______________________________________________

### CODE REVIEW

File: app-modules/service-management/src/Models/Scopes/ClassifiedIn.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+*/
+
+namespace AidingApp\ServiceManagement\Models\Scopes;
+
+use Illuminate\Database\Eloquent\Builder;
+
+class ClassifiedIn
+{
+    public function __construct(
+        protected array $classification,
```
Comment (Orrison) [inline app-modules/service-management/src/Models/Scopes/ClassifiedIn.php:44]: Let's make this plural, `classifications`

File: app-modules/service-management/src/Models/Scopes/ClassifiedIn.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Models\Scopes;
+
+use Illuminate\Database\Eloquent\Builder;
+
+class ClassifiedIn
+{
+    public function __construct(
```
Comment (Orrison) [inline app-modules/service-management/src/Models/Scopes/ClassifiedIn.php:43]: Are we able to add a DOCBLOCK here to specify the `$classifications` parameter should be an array of classifications?

Right now I think we have two classification ENUMs `SystemChangeRequestClassification` and `SystemServiceRequestClassification`.

So we can either make the DOCBLOCK typehint something like `SystemServiceRequestClassification[]|SystemServiceRequestClassification[].
Or better yet, we make a `Classification` interface, apply it to both classification classes and then make the typehint `Classification[]`

File: app/Filament/Widgets/ServiceRequestWidget.php [hunk 3]
```diff
@@ -80,7 +80,12 @@ protected function getStats(): array
 
     private function calculateOpenServiceRequestStats(Carbon $intervalStart): array
     {
-        $openStatusIds = ServiceRequestStatus::tap(new ClassifiedAs(SystemServiceRequestClassification::Open))->pluck('id');
+        $openStatusIds = ServiceRequestStatus::tap(new ClassifiedIn([
+            SystemServiceRequestClassification::Open,
+            SystemServiceRequestClassification::InProgress,
+            SystemServiceRequestClassification::Waiting,
+            SystemServiceRequestClassification::Custom,
```
Comment (Orrison) [inline app/Filament/Widgets/ServiceRequestWidget.php:87]: What do you think about returning this list from a static method of `SystemServiceRequestClassification` like `getUnclosedClassifications`.

That way if we ever add a new classification that would technically be a part of this list, we don't have to track down everywhere we put it.

**Review by Orrison — CHANGES_REQUESTED** [review]
Looks great, just a few comments and notes

File: app/Filament/Widgets/ServiceRequestWidget.php (hunk 3 above)
Comment (ankit-canyon) [reply app/Filament/Widgets/ServiceRequestWidget.php]: Yep that's been added !

Thank you.

File: app-modules/service-management/src/Models/Scopes/ClassifiedIn.php (hunk 2 above)
Comment (ankit-canyon) [reply app-modules/service-management/src/Models/Scopes/ClassifiedIn.php:48]: Yep that's been added !

Thank you.

File: app-modules/service-management/src/Models/Scopes/ClassifiedIn.php (hunk 1 above)
Comment (ankit-canyon) [reply app-modules/service-management/src/Models/Scopes/ClassifiedIn.php]: Yep that's been added !

Thank you.

**Review by danharrin — APPROVED** [review]

**Review by Orrison — APPROVED** [review]

**Pull request merged**